### DISCUSSION TITLE
Announcing Sionna v0.19

### DISCUSSION THREAD

**Discussion by jhoydis** [NVlabs/sionna#605] [Announcements]
# Ray Tracing

- New notebook: [Tutorial on Coverage Maps](https://nvlabs.github.io/sionna/examples/Sionna_Ray_Tracing_Coverage_Map.html).
  It computes coverage maps for different precoding vectors, visualizes metrics like path gain, received signal strength (RSS), and signal-to-interference-plus-noise ratio (SINR), estimates user-to-transmitter associations, and samples user positions for the generation of channel impulse responses
- [Scene](https://nvlabs.github.io/sionna/api/rt.html#sionna.rt.Scene) extended with [bandwidth](https://nvlabs.github.io/sionna/api/rt.html#sionna.rt.Scene.bandwidth) and [temperature](https://nvlabs.github.io/sionna/api/rt.html#sionna.rt.Scene.temperature) properties
- [Transmitter](https://nvlabs.github.io/sionna/api/rt.html?highlight=transmitter#sionna.rt.Transmitter) extended with a transmit power property in [Watt]( https://nvlabs.github.io/sionna/api/rt.html?highlight=transmitter#sionna.rt.Transmitter.power) and [dBm]( https://nvlabs.github.io/sionna/api/rt.html?highlight=transmitter#sionna.rt.Transmitter.power_dbm)
- [CoverageMap](
  https://nvlabs.github.io/sionna/api/rt.html?highlight=coveragemap#sionna.rt.CoverageMap) extended to compute [SINR maps](https://nvlabs.github.io/sionna/api/rt.html?highlight=coveragemap#sionna.rt.CoverageMap.sinr) and [RSS maps](https://nvlabs.github.io/sionna/api/rt.html?highlight=coveragemap#sionna.rt.CoverageMap.rss), in addition to the already existing [path gain maps](https://nvlabs.github.io/sionna/api/rt.html?highlight=coveragemap#sionna.rt.CoverageMap.path_gain)
- [CoverageMap]( https://nvlabs.github.io/sionna/api/rt.html?highlight=coveragemap#sionna.rt.CoverageMap) extended to [compute]( https://nvlabs.github.io/sionna/api/rt.html?highlight=coveragemap#sionna.rt.CoverageMap.cell_to_tx) and [visualize]( https://nvlabs.github.io/sionna/api/rt.html?highlight=coveragemap#sionna.rt.CoverageMap.show_association) cell-to-transmitter association
- [CoverageMap]( https://nvlabs.github.io/sionna/api/rt.html?highlight=coveragemap#sionna.rt.CoverageMap) extended to [compute and visualize the CDF]( https://nvlabs.github.io/sionna/api/rt.html?highlight=coveragemap#sionna.rt.CoverageMap.cdf) of path gain, RSS, or SINR
- [show()](https://nvlabs.github.io/sionna/api/rt.html?highlight=coveragemap#sionna.rt.CoverageMap.show) property of [CoverageMap](https://nvlabs.github.io/sionna/api/rt.html?highlight=coveragemap#sionna.rt.CoverageMap) extended to visualize SINR and RSS maps in addition to the existing path gain
  maps

- Adds feature to [Scene.coverage_map]( https://nvlabs.github.io/sionna/api/rt.html#sionna.rt.Scene.coverage_map) to execute multiple runs of shoot-and-bounce to compute coverage maps. This allows computing coverage maps using large number of rays despite memory limitation

- **Breaking Change**: `CoverageMap.as_tensor()` replaced by [path_gain]( https://nvlabs.github.io/sionna/api/rt.html?highlight=coveragemap#sionna.rt.CoverageMap.path_gain), [rss]( https://nvlabs.github.io/sionna/api/rt.html?highlight=coveragemap#sionna.rt.CoverageMap.rss), and [sinr]( https://nvlabs.github.io/sionna/api/rt.html?highlight=coveragemap#sionna.rt.CoverageMap.sinr) properties

# Config

- Pseudo-random generators can be configured from the [config](https://nvlabs.github.io/sionna/api/config.html) module to ensure reproducible results, for [Python](https://nvlabs.github.io/sionna/api/config.html#sionna.Config.py_rng), [Numpy](https://nvlabs.github.io/sionna/api/config.html#sionna.Config.np_rng) and [TensorFlow](https://nvlabs.github.io/sionna/api/config.html#sionna.Config.tf_rng) 

- New [config.seed property](https://nvlabs.github.io/sionna/api/config.html#sionna.Config.seed) to configure all Python, Numpy and TensorFlow pseudo-random generator seeds at once

# OFDM

- Adds feature to prepend cyclic prefixes to OFDM symbols of different lengths (see [OFDMModulator](https://nvlabs.github.io/sionna/api/ofdm.html?highlight=ofdmmodulator#sionna.ofdm.OFDMModulator) and the related PR [#465](https://github.com/NVlabs/sionna/pull/465))

# Precoding

- Adds feature to produce Discrete Fourier Transform (DFT) grid of beams for [uniform linear](https://nvlabs.github.io/sionna/api/mimo.html#sionna.mimo.grid_of_beams_dft_ula) and [rectangular](https://nvlabs.github.io/sionna/api/mimo.html#sionna.mimo.grid_of_beams_dft) MIMO antenna arrays

# Fixes

- Addresses the significant increase in memory footprint for path and coverage map computations introduced in version 0.18

- Fixes the skewness of the UTs position distribution toward the center of the sector generated via [drop_uts_in_sector](https://nvlabs.github.io/sionna/api/channel.wireless.html?highlight=drop_uts_in_sector#sionna.channel.drop_uts_in_sector), ensuring a uniform distribution across the sector

- Fixes ray leakage in coverage map computation (#564)

- Fixes the application of precoding vector in coverage map computation in the presence of multiple transmitters

- Fixes object IDs issues in ray tracer: Scene objects now always have contiguous indices starting from 0

- Fixes missing scenes in Docker (#597)

**Comment by fighting-xq** [comment]
Great Job
